core/src/main/java/io/grpc/internal/AbstractClientStream.java

```diff
@@ -104,6 +104,9 @@ protected AbstractClientStream(
       Metadata headers,
       boolean useGet) {
     Preconditions.checkNotNull(headers, "headers");
+    if (transportTracer != null) {
+      transportTracer.reportStreamStarted();
```

Comment:
Should this be done in the transportstate instead?

Reply:
`onStreamAllocated` looks like a good place to move this